partially (re-)implemented control messages

damanmkumar opened the pull request:
- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages IO channels around a socket
- serializer classes manage serialization based on recipient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to `comm` namespace
- `inet` namespace renamed to `comm`
- `example/proto-msg` only serializes one message for now

damanmkumar committed:
control messages - v2 -- WIP

damanmkumar committed:
control messages - v2 -- WIP

damanmkumar committed:
control messages - v2 -- WIP

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
partially (re-)implemented control messages

- wrapper classes for protobufs only provide an interface
  to create control messages
- channel class manages io channels around a socket
- serializer classes manage serialization based on recepient
- separation of responsibilities done so that new control
  messages can be added without modifying existing code
- socket class moved to comm namespace
- "inet" namespace renamed to "comm"
- example/proto-msg only serializes one message for now

damanmkumar committed:
Merge branch 'dev' into feat/ctl-msgs-alt

damanmkumar committed:
fixed wrong scoping of service + explicitly scoped
the type and value of {file|chunk}_request::msg_type

damanmkumar committed:
fixed wrong scoping of service + explicitly scoped
the type and value of {file|chunk}_request::msg_type

damanmkumar committed:
fixed wrong scoping of service + explicitly scoped
the type and value of {file|chunk}_request::msg_type

damanmkumar committed:
removed unused files

damanmkumar committed:
removed unused files

damanmkumar committed:
removed unused files

damanmkumar committed:
removed unused file

damanmkumar committed:
removed unused file

damanmkumar committed:
removed unused file

abmono93 committed:
Create sprint-04.md

abmono93 committed:
Added Adam's tasks to spring backlog

damanmkumar committed:
answered sprint retrospective questions

damanmkumar committed:
made further changes to the report

damanmkumar committed:
fixed typo

abmono93 committed:
Minor formatting

damanmkumar committed:
renamed {file|chunk}_request::msg_type to type

damanmkumar committed:
renamed {file|chunk}_request::msg_type to type

damanmkumar committed:
renamed {file|chunk}_request::msg_type to type

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
implemented support for sending messages to chunk servers

damanmkumar committed:
Merge branch 'feat/ctl-msgs-alt' of github.com:CS297P-SAD/sadfs into feat/ctl-msgs-alt

damanmkumar committed:
added deserialization support

damanmkumar committed:
added deserialization support

abmono93 commented on src/proto/master.proto:
```diff
@@ -0,0 +1,20 @@
+	io_type type     = 1;
+	string  filename = 2;
+	uint64  offset   = 3;
+	uint64  length   = 4;
```
Is this meant to be the offset and length measured in bytes or chunks?

abmono93 commented on include/sadfs/msgs/messages.hpp:
```diff
@@ -0,0 +1,106 @@
+	std::string const filename;
+	std::size_t const offset;
+	std::size_t const length;
+};
```
I guess that answers my other question about the protobuf definition.
I'm not against doing it this way but my understanding was that this logic would be taken care of on the client side and the servers would deal with chunks of data. i.e. it would be the client's responsibility to know what chunk it needs to access and send a request accordingly, e.g. it asks for the third chunk of file `main.cpp` and the master server will tell it the chunk's id and location. From what I can see this would be a simpler approach. We can talk about this elsewhere if it will be a long conversation

shu-g committed:
Created sadfuse class and sadfsd-bootstrap (#30)

* Created sadfuse class and sadfsd-bootstrap
* Implemented a bare bones fuse wrapper
* Modified CMakeFile to support libfuse
* Added sadfsd-bootstrap file

* Attempt #2 at modifying Github actions to support FUSE

* Attempt #3 at modifying Github actions to support FUSE

* Attempt #4 at modifying Github actions to support FUSE

* Attempt #5 at modifying Github actions to support FUSE

* Attempt #6 at modifying Github actions to support FUSE

* Attempt #7 at modifying Github actions to support FUSE

* Attempt #8 at modifying Github actions to support FUSE

* Attempt #9 at modifying Github actions to support FUSE

* Attempt #10 at modifying Github actions to support FUSE

* Attempt #11 at modifying Github actions to support FUSE

* Attempt #12 at modifying Github actions to support FUSE

* Attempt #13 at modifying Github actions to support FUSE

* Attempt #14 at modifying Github actions to support FUSE

* Attempt #15 at modifying Github actions to support FUSE

* Update ideas.md

* Update ideas.md

* Attempt #16 at modifying Github actions to support FUSE

* Changed sadfuse from singleton to regular class.
* Modified operations_ to static member of sadfuse
* Replaced socket with inet::service in sadfuse
* Made formatting corrections

* Modified sadfsd-bootstrap build in the CMakeFile to use the foreach loop

* Changed sadfuse class to sadfilesys.
* Modified master_service_ to static shared pointer

* Added this_() function to the class sadfilesys()

* Changed operations to non-static methods

* Miscellaneous changes
* Deleted sadfuse files
* Removed extraneous headers
* Renamed sadfilesys::run() to bootstrap()

* updated a comment, and fixed intermixing of tabs and spaces (#49)

* updated a comment

* fixed intermixing of tabs and spaces

* fix build error due to missing dependency config (#50)

Co-authored-by: damanmkumar <59807165+damanmkumar@users.noreply.github.com>

damanmkumar committed:
renamed inet namespace to comm (#48)

* renamed inet namespace to comm

- implemented the move assignment operator in the socket
  class
- fixed early termination bug in echod

* Created sadfuse class and sadfsd-bootstrap (#30)

* Created sadfuse class and sadfsd-bootstrap
* Implemented a bare bones fuse wrapper
* Modified CMakeFile to support libfuse
* Added sadfsd-bootstrap file

* Attempt #2 at modifying Github actions to support FUSE

* Attempt #3 at modifying Github actions to support FUSE

* Attempt #4 at modifying Github actions to support FUSE

* Attempt #5 at modifying Github actions to support FUSE

* Attempt #6 at modifying Github actions to support FUSE

* Attempt #7 at modifying Github actions to support FUSE

* Attempt #8 at modifying Github actions to support FUSE

* Attempt #9 at modifying Github actions to support FUSE

* Attempt #10 at modifying Github actions to support FUSE

* Attempt #11 at modifying Github actions to support FUSE

* Attempt #12 at modifying Github actions to support FUSE

* Attempt #13 at modifying Github actions to support FUSE

* Attempt #14 at modifying Github actions to support FUSE

* Attempt #15 at modifying Github actions to support FUSE

* Update ideas.md

* Update ideas.md

* Attempt #16 at modifying Github actions to support FUSE

* Changed sadfuse from singleton to regular class.
* Modified operations_ to static member of sadfuse
* Replaced socket with inet::service in sadfuse
* Made formatting corrections

* Modified sadfsd-bootstrap build in the CMakeFile to use the foreach loop

* Changed sadfuse class to sadfilesys.
* Modified master_service_ to static shared pointer

* Added this_() function to the class sadfilesys()

* Changed operations to non-static methods

* Miscellaneous changes
* Deleted sadfuse files
* Removed extraneous headers
* Renamed sadfilesys::run() to bootstrap()

* updated a comment, and fixed intermixing of tabs and spaces (#49)

* updated a comment

* fixed intermixing of tabs and spaces

* fix build error due to missing dependency config (#50)

Co-authored-by: damanmkumar <59807165+damanmkumar@users.noreply.github.com>

* renamed inet namespace to comm

- implemented the move assignment operator in the socket
  class
- fixed early termination bug in echod

* removed explicit sets of function pointers to nullptr, for portability

Co-authored-by: shu-g <59781806+shu-g@users.noreply.github.com>

damanmkumar committed:
Merge branch 'feat/ctl-msgs-alt' of github.com:CS297P-SAD/sadfs into feat/ctl-msgs-alt

damanmkumar commented on src/proto/chunk.proto:
```diff
@@ -0,0 +1,18 @@
+message control_message {
+	sadfs.proto.host_id sender = 1;
+	oneof               msg {
+		chunk_request chunk_req = 11;
```
These are IDs that protobufs use internally to identify the fields that are being sent/received over the network. Changing these IDs is apparently not a good idea for backward compatibility. So, I left the first 10 for other fields we might need to add later.
